File: solutions/prolog/space-age/1/space_age.pl
space_age("Mercury", AgeSec, Years) :-
    age_on_earth(AgeSec / 0.2408467, Years).

space_age("Venus", AgeSec, Years) :-
    age_on_earth(AgeSec / 0.61519726, Years).

space_age("Earth", AgeSec, Years) :-
    age_on_earth(AgeSec, Years).

space_age("Mars", AgeSec, Years) :-
    age_on_earth(AgeSec / 1.8808158, Years).

space_age("Jupiter", AgeSec, Years) :-
    age_on_earth(AgeSec / 11.862615, Years).

space_age("Saturn", AgeSec, Years) :-
    age_on_earth(AgeSec / 29.447498, Years).

space_age("Uranus", AgeSec, Years) :-
    age_on_earth(AgeSec / 84.016846, Years).

space_age("Neptune", AgeSec, Years) :-
    age_on_earth(AgeSec / 164.79132, Years).

age_on_earth(AgeSec, Years) :-
    Years is (AgeSec / 31557600).

File: solutions/c/space-age/1/space_age.c
#include "space_age.h"

float age(planet_t planet, int64_t seconds)
{
    double age_earth = seconds / 31557600.0;
    
    switch (planet)
    {
        case EARTH:
            return age_earth;
        case MERCURY:
            return age_earth / 0.2408467;
        case VENUS:
            return age_earth / 0.61519726;
        case MARS:
            return age_earth / 1.8808158;
        case JUPITER:
            return age_earth / 11.862615;
        case SATURN:
            return age_earth / 29.447498;
        case URANUS:
            return age_earth / 84.016846;
        case NEPTUNE:
            return age_earth / 164.79132;
        default:
            return -1;
    }
}